httpmw/ip_cognito_auth.go

```diff
@@ -113,9 +115,10 @@ func (k *Key) RSA256() (*rsa.PublicKey, error) {
 // IpCognitoAuth middleware for:
 // * IP ranges verification in format "192.168.10.1-192.168.10.10,192.168.90.1-192.168.90.10"
 // * cognito authentication through Authorization Bearer Token
-func IpCognitoAuth(ipRange string, svc cognitoidentityprovideriface.CognitoIdentityProviderAPI, clientID string) gin.HandlerFunc {
+func IpCognitoAuth(ipRange string, svc cognitoidentityprovideriface.CognitoIdentityProviderAPI, clientID string, expire time.Duration) gin.HandlerFunc {
 	ipRanges := getIpRanges(ipRange)
 	jwk := new(JWK)
+	ch := cache.New(expire, expire)
```

Comment:
it's neccesary to check if expire is >= 1, because:
`If the expiration duration is less than one (or NoExpiration), the items in the cache never expire (by default), and must be deleted manually. If the cleanup interval is less than one, expired items are not deleted from the cache before calling c.DeleteExpired().`

Reply:
I think that instead of adding another condition we should just add this to the method description.